### ISSUE TITLE
React: Leaving IBM - There is no label for the modal dialog

### ISSUE THREAD

**Issue by andysherman2121** [carbon-design-system/carbon-for-ibm-dotcom#5775]
On JAWS FF/Voiceover iPhone
Issue:
There is no label for the modal dialog

Expected:
Provide the heading present in the dialog as the label by associating the same using aria-labelledby

**Comment by andysherman2121** [comment]
@Praveen-sr can you retest this?

**Comment by Praveen-sr** [comment]
@andysherman2121 Ram re tested this on react canary, now on JAWS FF its fixed and working but on iPhone Voiceover the issue still exits.
